python/tach/sync.py

```diff
@@ -18,52 +19,50 @@ def sync_dependency_constraints(
     project_root: Path,
     project_config: ProjectConfig,
     exclude_paths: list[str] | None = None,
+    prune: bool = True,
 ) -> ProjectConfig:
     """
     Update project configuration with auto-detected dependency constraints.
-    This is additive, meaning it will create dependencies to resolve existing errors,
+    If prune is set to False, it will create dependencies to resolve existing errors,
     but will not remove any constraints.
     """
+    if prune:
+        # Create a blank config
+        new_config = project_config.model_copy(
+            update={
+                "modules": [
+                    module.model_copy(update={"depends_on": []})
+                    for module in project_config.modules
+                ]
+            }
+        )
+        # Update deprecations first
+        check_result = check(
+            project_root=project_root,
+            project_config=project_config,
+            exclude_paths=exclude_paths,
+        )
```

Comment:
This is an expensive call, I think it'll noticeably slow down `sync` on larger codebases. If I understand correctly, the intention is to figure out which deprecated dependencies are actually causing warnings, and preserve those. Instead of running `check` here, could you instead record a mapping of the existing deprecated dependencies, and then add them back to the config if they appear as full-blown errors in the check below?